rinex/src/navigation/orbits.rs

```diff
@@ -230,175 +226,191 @@ impl OrbitItem {
-            //     -> item is unique (if dB declaration is correct)
-            //     -> return directly
-            let major = u8::from_str_radix(minor_matches[0].major, 10).unwrap();
-            let minor = u8::from_str_radix(minor_matches[0].minor, 10).unwrap();
-            Some(version::Version::new(major, minor))
-        } else {
-            // [+] .major perfectly matched
-            // [+] .minor not perfectly matched
-            //    --> use closest older minor revision
-            let mut to_sort: Vec<_> = major_matches
-                .iter()
-                .map(|r| {
-                    (
-                        u8::from_str_radix(r.major, 10).unwrap(), // to later build object
-                        u8::from_str_radix(r.minor, 10).unwrap(), // to later build object
-                        i8::from_str_radix(r.minor, 10).unwrap() - desired_rev.minor as i8, // for filter op
-                    )
-                })
-                .collect();
-            to_sort.sort_by(|a, b| b.2.cmp(&a.2)); // sort by delta value
-            let to_sort: Vec<_> = to_sort
-                .iter()
-                .filter(|r| r.2 < 0) // retain negative deltas : only older revisions
-                .collect();
-            Some(version::Version::new(to_sort[0].0, to_sort[0].1))
-        }
-    } else {
-        // ---> desired_rev.major not perfectly matched
-        // ----> use closest older major revision
-        let mut to_sort: Vec<_> = revisions
+/*
+ * Identifies closest (but older) revision contained in NAV database.
+ * Closest content (in time) is used during record parsing to identify and sort data.
+ * Returns None
+ *   - if no database entries were found for requested constellation.
+ *   - or only newer revision exist : we prefer matching on older revisions
+ */
+pub(crate) fn closest_nav_standards(
+    constellation: Constellation,
+    revision: version::Version,
+    msg: NavMsgType,
+) -> Option<&'static NavHelper<'static>> {
+    let database = &NAV_ORBITS;
+    // start by trying to locate desired revision.
+    // On each mismatch, we decrement and try to match.
+    let (mut major, mut minor): (u8, u8) = revision.into();
+    loop {
+        // filter on both:
+        //  + Exact Constellation
+        //  + Exact NavMsgType
+        //  + Exact revision we're currently trying to locate
+        //    algorithm: decreasing, starting from desired revision
+        let items: Vec<_> = database
             .iter()
-            .map(|r| {
-                (
-                    u8::from_str_radix(r.major, 10).unwrap(), // to later build object
-                    i8::from_str_radix(r.major, 10).unwrap() - desired_rev.major as i8, // for filter op
-                    u8::from_str_radix(r.minor, 10).unwrap(), // to later build object
-                    i8::from_str_radix(r.minor, 10).unwrap() - desired_rev.minor as i8, // for filter op
-                )
+            .filter_map(|item| {
```

Comment:
This could be `.filter(|item| item.constellation == constellation && item.msg == msg && item.version == Version::new(major, minor)`, right?

Reply:
correct! applied